packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -69,18 +71,9 @@ class MacOSAssetBehavior extends SourceBehavior {
 /// Removes any previous version of the framework that already exists in the
 /// target directory.
 // TODO(jonahwilliams): remove shell out.
-class UnpackMacOS extends Target {
+abstract class UnpackMacOS extends Target {
```

Comment:
I think it would be useful to update the comment to explain why it's abstract and what the subclasses are for (with references to those classes for convenience of searching).

Reply:
Added more documentation on the class, and a todo linking to an issue for cleaning it up with further assemble improvements.